.github/workflows/build.yml

```diff
@@ -19,6 +19,9 @@ jobs:
         pip install -r requirements.txt -r requirements-dev.txt
     - name: Run pre-commit
       uses: pre-commit/action@v2.0.0
+    - name: Check types with mypy
+      run: |
+        mypy --strict .
```

Comment:
I found that when mypy fails on CI, it's useful to get html report via `--html-report`, I don't know whether that would work well with github actions tho 🤔 ?

Reply:
That would be very useful! However, I'm not sure it's currently possible with GH actions, see https://github.community/t/publishing-test-results/16215/11. I suggest we just leave it as the text report.

Reply:
@tomwhite oh, that not great. Anyway makes sense if that's the limitation of GH actions.